test/hotspot/jtreg/vmTestbase/nsk/share/jdb/JdbTest.java

```diff
@@ -204,37 +204,34 @@ protected int runTest(String argv[], PrintStream out) {
                     }
                 }
 
-            } catch (Exception e) {
-                failure("Caught unexpected exception: " + e);
-                e.printStackTrace(out);
-
+            } catch (Throwable t) {
+                failure("Caught unexpected exception: " + t);
+                t.printStackTrace(out);
+            } finally {
                 if (jdb != null) {
+                    log.complain("jdb reference is not null, check for exception in the logs.");
                     try {
                         jdb.close();
                     } catch (Throwable ex) {
                         failure("Caught exception/error while closing jdb streams:\n\t" + ex);
                         ex.printStackTrace(log.getOutStream());
                     }
-                } else {
-                    log.complain("jdb reference is null, cannot run jdb.close() method");
                 }
 
-                if (debuggee != null) {
+                if (debuggee != null && debuggee.terminated()) {
```

Comment:
Shouldn't this be `!debuggee.terminated()`?

Reply:
Thanks for catching it! I've updated check.